### PR TITLE
[lex] Replace 'could' and 'might'

### PR DESCRIPTION
**Pull request by jensmaurer**
as directed by ISO/CS.

Partially addresses #4319

### CODE REVIEW

File: source/lex.tex [hunk 1]
```diff
@@ -401,7 +401,7 @@
 preprocessing number token (one that is not a valid
 \grammarterm{integer-literal} or \grammarterm{floating-point-literal} token),
 even though a parse as three preprocessing tokens
-\tcode{0xe}, \tcode{+}, and \tcode{foo} might produce a valid expression (for example,
+\tcode{0xe}, \tcode{+}, and \tcode{foo} can produce a valid expression (for example,
 if \tcode{foo} were a macro defined as \tcode{1}). Similarly, the
```
Comment (tkoeppe) [inline source/lex.tex:405]: If we keep "can" here, then I think the parenthetical needs to say "is", not "were".

Alternatively, could we say "would be able to produce"?

File: source/lex.tex [hunk 2]
```diff
@@ -362,7 +362,7 @@
 
 \item Otherwise,
 the next preprocessing token is the longest sequence of
-characters that could constitute a preprocessing token, even if that
+characters that would constitute a preprocessing token, even if that
```
Comment (tkoeppe) [inline source/lex.tex:365]: I'm not sure "would" works here. Second options? @zygoloid?

File: source/lex.tex [hunk 3]
```diff
@@ -1893,7 +1893,7 @@
 \end{example}
 The syntactic non-terminal preceding the \grammarterm{ud-suffix} in a
 \grammarterm{user-defined-literal} is taken to be the longest sequence of
-characters that could match that non-terminal.
+characters that would match that non-terminal.
```
Comment (tkoeppe) [inline source/lex.tex:1896]: (Same here as above. If the above is OK, so is this.)

File: source/lex.tex (hunk 2 above)
Comment (jensmaurer) [reply source/lex.tex:365]: We have a bunch of max-munch rules that all follow the same pattern.

Comment (tkoeppe) [reply source/lex.tex:365]: OK, then maybe this is fine as is (likewise below).

Comment (zygoloid) [reply source/lex.tex:365]: What I've been suggesting in other PRs is:
"longest sequence of characters that matches the syntax of a *preprocessing-token*"
... which seems more precise than our prior wording in the various tentative parsing areas, and distinguishes between matching the syntax of X and actually forming a valid X, without getting into hypotheticals.

File: source/lex.tex (hunk 1 above)
Comment (zygoloid) [reply source/lex.tex:405]: I'm happy with either. I'm a little worried that ISO will decide to declare "would" as verboten next (the upcoming revised Directives disallow use of any verbal forms not explicitly permitted, which would disallow "would"). But we can face that if and when it happens.

File: source/lex.tex (hunk 3 above)
Comment (zygoloid) [reply source/lex.tex:1896]:
```suggestion
characters that match the syntax of that non-terminal.
```

File: source/lex.tex (hunk 2 above)
Comment (jensmaurer) [reply source/lex.tex:365]: fixed

File: source/lex.tex (hunk 1 above)
Comment (jensmaurer) [reply source/lex.tex:405]: fixed to "is"

We need to be able to talk about hypotheticals in proper English phrasing, that includes "could" and "were" and "would" and "might". Everything else is just stupid.
(I agree with some of the -> can changes we have been doing; they seem to be clear improvements occasionally.)

**Pull request merged**